server/resources/resource_management.go

```diff
@@ -1166,6 +1167,66 @@ func ActionResource(context middleware.Context, resourceSchema *schema.Schema,
 	return fmt.Errorf("no response")
 }
 
+func checkIfActionIsAllowedForUser(context middleware.Context, dataStore db.DB, dataMap map[string]interface{},
```

Comment:
If a user without access to a resource, but with knowledge of its ID, calls an action on it, we correctly reject such attempts, but instead of returning 404, we return 401. That betrays the fact that such resource exists.